src/main/java/seedu/cakecollate/model/orderitem/Type.java

```diff
@@ -0,0 +1,56 @@
+package seedu.cakecollate.model.orderitem;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.cakecollate.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the type of cake of an order item.
+ * Guarantees: immutable; is valid as declared in {@link #isValidType(String)}
+ */
+public class Type {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Order Type should only contain alphabets and white spaces, and it should not be blank";
+
+    /**not sure if there is a way to simplify this regex? It matches anything that
+     * contains only alphabets and does not start or end with a white space*/
+    public static final String VALIDATION_REGEX = "^([a-zA-Z]|([a-zA-Z]+[a-zA-Z\\s]*[a-zA-Z]))$";
```

Comment:
Maybe the Name class contains the regex you're looking for? You could adapt the "\\p{Alnum}" in the Name regex to \\p{Alpha} . I'm basing that off this link (https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html) 

Also, I think when the parser part is done, it should be trimming any user input before the Type constructor is called so checking for spaces may not be an issue?

Reply:
Oh yeah good point about the parser! I will edit the regex accordingly.

I think the regex in the Name class does not accept single characters.